mongo/database.go

```diff
@@ -90,3 +91,21 @@ func (db *Database) Drop(ctx context.Context) error {
 	}
 	return nil
 }
+
+// ListCollections list collections from mongodb database.
+func (db *Database) ListCollections(ctx context.Context, options []options.ListCollectionsOptioner, filter *bson.Document) (command.Cursor, error) {
```

Comment:
Let's swap the options and filter arguments and make options variadic so users can choose if they want to provide options.